### ISSUE TITLE
Date Formatting

### ISSUE THREAD

**Issue by refringe** [sp-tarkov/forge#18]
---
### Attachments
- [Screenshot_20240905-215919.png](https://dev.sp-tarkov.com/attachments/56f10462-5540-4348-ab21-09b925238345) (106 KiB)

---

[Originally written](https://dev.sp-tarkov.com/SPT/forge/issues/45) by _**daz96050**_

**Comment by refringe** [comment]
I'd like there to be a few different states to dates, if possible:

- A close date should show relatively ("X minutes ago")
    - Bonus points for getting dynamically updated
- Dates over a month old (?) should show the month, day, and 12h time
- Dates within a different year should show the month, day, year

The datetime should

- Be converted to the current user's timezone, if available.
- Be available in the `time` tag's `datetime` attribute:
    ```html
    <time datetime="YYYY-MM-DDThh:mm:ssTZD">3 minutes ago</time>
    ```

There should be a [Blade directive](https://laravel.com/docs/11.x/blade#extending-blade) created for this to ensure the views remain clean of as much logic as possible.

There will need to be a way to fetch the user's timezone. There's a few ways we can do this, the easiest may just be to use UTC for all guests and save a profile option for a user's timezone. We could also look at using a front-end library to detect the date and send it to the back-end to save in the session... but I'm not sure the extra complexity would be worth it.

---
[Originally written](https://dev.sp-tarkov.com/SPT/forge/issues/45#issuecomment-28107) by _**Refringe**_

**Comment by refringe** [comment]
Resolved in 75ebc827dc1976b65ae64eb764bba0366a74b8e2. Currently in `develop`.

---
[Originally written](https://dev.sp-tarkov.com/SPT/forge/issues/45#issuecomment-28312) by _**Refringe**_
